x-pack/plugin/kql/src/test/java/org/elasticsearch/xpack/kql/parser/KqlParserTests.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.kql.parser;
+
+import org.elasticsearch.index.query.MatchAllQueryBuilder;
+import org.elasticsearch.index.query.SearchExecutionContext;
+import org.elasticsearch.test.AbstractBuilderTestCase;
+
+import java.io.IOException;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.isA;
+
+public class KqlParserTests extends AbstractBuilderTestCase {
+
+    public void testEmptyQueryParsing() throws IOException {
+        KqlParser parser = new KqlParser();
+        SearchExecutionContext searchExecutionContext = createSearchExecutionContext();
+        assertThat(parser.parseKqlQuery("", searchExecutionContext), isA(MatchAllQueryBuilder.class));
+    }
+
+    public void testSyntaxErrorsHandling() throws IOException {
```

Comment:
ℹ️ My intent is not to be exhaustive here but to test few query with errors.
 I will add more cases in ad-hoc test methods.

Reply:
I guess we need some testing to ensure the grammar works as expected, with KQL examples? Tests are necessary to ensure changes to the grammar (if needed) do not break existing cases. 

How are you thinking in terms of testing?

Reply:
The next PR will be adding the implementation of the `KqlAstBuilder` that will be building the QueryBuilder from the parsing tree. My plan is to add more tests cases in parallel of the query builder, so it becomes more exhaustive.

As an example, for the AND operator, I will probably be adding the following tests:

1. Basic nominal test:

check `parser. parseKqlQuery("foo AND mapped_field: bar")` generates the boolean query

2. Adding some error syntax 

- check `parser. parseKqlQuery("AND mapped_field: bar")` throws

- check `parser. parseKqlQuery("foo AND")` throws

- check `parser. parseKqlQuery("foo AND )")` throws

- ...

2. Specifically for boolean operators:

 - some precedence tests
```
parser. parseKqlQuery("foo AND bar OR baz") == parser. parseKqlQuery("foo AND (bar OR baz)")
```

  - check they are not case sensitive
```
parser. parseKqlQuery("foo AND bar") == parser. parseKqlQuery("foo and bar")
```

  - check they escapement
```
parser. parseKqlQuery("foo AND bar") == parser. parseKqlQuery("foo and bar")
```

Reply:
Makes sense, thanks for explaining!